core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchModelImpl.java

```diff
@@ -218,11 +206,19 @@ public String getId() {
     }
 
     private Optional<String> createAutocompleteUrl() {
-        return Optional.ofNullable(request).map(r -> pathTransformer.map(r, resource.getPath()))
-                .map(url -> String.format("%s.%s.%s",
-                        url,
-                        AutocompleteServlet.AUTOCOMPLETE_SELECTOR,
-                        AutocompleteServlet.EXTENSION_JSON));
+        if (Optional.ofNullable(searchCAConfigurationModel)
+                    .filter(SearchCAConfigurationModel::isAutocompleteEnabled)
+                    .isPresent()) {
+            return Optional.ofNullable(request)
+                           .map(r -> pathTransformer.map(r, resource.getPath()))
+                           .map(url -> String.format("%s.%s.%s",
+                                                     url,
+                                                     AutocompleteServlet.AUTOCOMPLETE_SELECTOR,
+                                                     AutocompleteServlet.EXTENSION_JSON));
+        } else {
```

Comment:
You can remove the else, we will always end up here

Reply:
It could end up in the else, if the autocomplete disabled

Reply:
yes, I main to change to that:
```
if (...) {
...
}
log.info("Autocomplete is not enabled. To enable it, please check context aware SearchConfiguration.");
return Optional.empty();
```